lib/internal/test_runner/harness.js

```diff
@@ -29,25 +29,57 @@ const {
 const { queueMicrotask } = require('internal/process/task_queues');
 const { TIMEOUT_MAX } = require('internal/timers');
 const { clearInterval, setInterval } = require('timers');
+const { validatePath } = require('internal/fs/utils');
+const { pathToFileURL } = require('internal/url');
 const { bigint: hrtime } = process.hrtime;
 const resolvedPromise = PromiseResolve();
 const testResources = new SafeMap();
+const esmLoader = require('internal/modules/esm/loader');
+const { kEmptyObject } = require('internal/util');
+const { validateFunction } = require('internal/validators');
+const { getOptionValue } = require('internal/options');
 let globalRoot;
+let globalSetupFunction;
+let globalTeardownFunction;
+let globalSetupExecuted = false;
 
 testResources.set(reporterScope.asyncId(), reporterScope);
 
+async function setupGlobalSetupTeardownFunctions(globalSetupPath, cwd) {
```

Comment:
All of this might fit better in `utils.js`, and called from `parseCommandLine()`, similar to how some other flags like `--test-name-pattern` are processed.

Reply:
Regarding having it as part of `utils.js`: I agree, it definitely makes more sense.  
As for `parseCommandLine`: at the moment, it is not "`cwd`-aware", and I'm not sure whether it should have that dependency.